core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/RetryableStage.java

```diff
@@ -43,7 +43,7 @@ public RetryableStage(HttpClientDependencies dependencies,
                           RequestPipeline<SdkHttpFullRequest, Response<OutputT>> requestPipeline) {
         this.dependencies = dependencies;
         this.requestPipeline = requestPipeline;
-        this.rateLimitingTokenBucket = new RateLimitingTokenBucket();
+        this.rateLimitingTokenBucket = null;
```

Comment:
Since you're reviewing, @dagnir: Isn't the RetryableStage created with each request, so this rate limiting token bucket is never reused between requests, making it essentially worthless?

Reply:
I didn't fix that here. I can create a backlog item to fix it...